src/main/kotlin/no/nav/klage/pdfgen/service/ForlengetBehandlingstidService.kt

```diff
@@ -0,0 +1,203 @@
+                    }
+                    p {
+                        +"Om du ikke ønsker å logge deg inn på nav.no kan du kan du gå til nav.no/klage og trykke på \"Ettersend dokumentasjon\" for det saken gjelder."
+                    }
+                    h2 { +"Du har rett til innsyn" }
+                    p {
+                        +"Du har rett til å se dokumentene i saken din."
+                    }
+                    h2 { +"Informasjon om fri rettshjelp" }
+                    p {
+                        +"Dette får du vite mer om hos Statsforvalteren eller advokat."
+                    }
+                    div {
+                        classes = setOf("signature")
+                        +"Med hilsen"
+                        br { }
+                        +"Nav klageinstans"
+                    }
+                }
+            }
+    }
+
+    private fun getBehandlingstidText(forlengetBehandlingstidRequest: ForlengetBehandlingstidRequest): String {
+        if (forlengetBehandlingstidRequest.behandlingstidDate != null) {
+            return getFormattedDate(LocalDate.parse(forlengetBehandlingstidRequest.behandlingstidDate))
+        } else if (forlengetBehandlingstidRequest.behandlingstidUnits != null && forlengetBehandlingstidRequest.behandlingstidUnitTypeId != null) {
+            return forlengetBehandlingstidRequest.behandlingstidUnits.toString() + when (TimeUnitType.of(forlengetBehandlingstidRequest.behandlingstidUnitTypeId)) {
+                TimeUnitType.WEEKS -> {
+                    if (forlengetBehandlingstidRequest.behandlingstidUnits == 1) {
+                        " uke"
+                    } else {
+                        " uker"
+                    }
+                }
+
+                TimeUnitType.MONTHS -> {
+                    if (forlengetBehandlingstidRequest.behandlingstidUnits == 1) {
+                        " måned"
+                    } else {
+                        " måneder"
+                    }
+                }
+            }
+        } else {
+            throw Exception("Needs date or units and unit type")
+        }
+    }
+
+    private fun String.toSpecialCase(): String {
+        val strings = this.split(" - ")
+        return when (strings.size) {
+            1 -> {
+                this.decapitalize()
+            }
+
+            2 -> {
+                if (strings[0].equals(other = strings[1], ignoreCase = true)) {
+                    strings[0].decapitalize()
+                } else {
+                    strings[0].decapitalize() + " - " + strings[1].decapitalize()
+                }
+            }
+
+            else -> this
+        }
+    }
+
+    private fun String.decapitalize(): String {
+        return if (!this.startsWith("NAV")) {
+            this.replaceFirstChar(Char::lowercase)
+        } else this
+    }
+
+    private fun String.toFnrView() = this.substring(0, 6) + " " + this.substring(6)
```

Comment:
Er disse kopierte nå? Man kunne ev. satt de in en util-fil?

Reply:
Enig, skal fikse det.